### DISCUSSION TITLE
Unable to filter the localSharePointresult using the CreatedBy or author fields inside my PnP Modern Search web part

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#1555] [Q&A]
I am using the pnp-modern-search v4 and i have the following Search Result web part:-

![S1](https://user-images.githubusercontent.com/79726511/143293357-ceae8d99-3188-4535-85bf-b4e1c112d5ba.png)

which mainly searches everything inside a specific site collection.

Then i defined the following filter web part, to filter the results based on; Created + File type + Created By (Author), here is my filter web part's settings:-

![S2](https://user-images.githubusercontent.com/79726511/143293454-e2772499-4219-4af8-aa9f-067cf7fc67cc.png)

but when i save the page >> i will only get the Created + the File Type, while the Created By & Author filters will be missing, as follow:-
![s3](https://user-images.githubusercontent.com/79726511/143293494-a2339fac-6ac7-48ed-b753-7509af28ae4e.png)

also if i add a new filter web part and i only chose to filter based on the Created By i will get this error:-
![s4](https://user-images.githubusercontent.com/79726511/143293303-7c74a1a1-73f9-4c2d-969b-e1342065aef7.png)

any advice why i am unable to filter the search result using the CreatedBy/Author?

**Comment by FranckyC** [comment]
Hi @mvcsharepointdev, `CreatedBy` and `AuthorOWSUSER` are not refinable managed properties by default so can't be used as refiners. You must use `RefinableStringXX` properties in your search schema mapped to the corresponding crawled properties to make it work.

**Reply by mvcsharepointdev** [reply]
@FranckyC the weird thing is that on another tenant i were able to filter based on the CreatedBy and AuthorOWSUSER even they are not defined as Refinable as well.. so why we got 2 different behaviors on 2 different tenants? any idea? thanks

**Reply by wobba** [reply]
> @FranckyC the weird thing is that on another tenant i were able to filter based on the CreatedBy and AuthorOWSUSER even they are not defined as Refinable as well.. so why we got 2 different behaviors on 2 different tenants? any idea? thanks

That's a know internal Microsoft bug that it works on some tenants, and it doesn't really work 100%. You can expect that to stop working.

**Reply by mvcsharepointdev** [reply]
@wobba  ok noted, as it worked on our tenant, but did not work on the customer tenant.. so to be on the safe side, we should define a RefinableStringXX managed property and link it to the "ows_Created_x0020_By" crawl property?

**Reply by mvcsharepointdev** [reply]
@wobba  ok i edited the RefinableString00 to be linked to the "ows_Created_x0020_By" crawl property.. then i manually typed the RefinableString00 inside the "PnP-Search filter" web part..but the filter did not show inside the "PnP-Search filter" web part, and the related "PnP-Search Results" web part will always return 0 results... is this because i need to wait for sometime, to have the RefinableString00 working? till a full crawl is performed?

**Reply by wobba** [reply]
There is no full crawl happening - you need to manually trigger crawl of the sites/libs. And also read https://github.com/microsoft-search/pnp-modern-search/issues/1283#issuecomment-917671147 for what to map.

**Reply by mvcsharepointdev** [reply]
@wobba  thanks for the reply.. now in our tenant, i can not find a crawl property named "Office:4" to map it to refinable managed property... not sure why?
![text](https://user-images.githubusercontent.com/79726511/143431943-e05b576d-41d8-4b8e-a514-97bd05d0dba7.png)

**Reply by wobba** [reply]
Pick "Office" in the category.

**Reply by mvcsharepointdev** [reply]
@wobba Thanks for the reply, i tried the following but did not work:-

1) I mapped the Office to the RefinableString00 as follow:-
![rrrrrrrrrrr1](https://user-images.githubusercontent.com/79726511/143480881-316f9565-8816-401d-8067-216ef4c2c8a9.png)


2) I added the RefinableString00 inside the ""PnP-Searh filter" we part:-
![eeeeeeeeeeeeeeeeeeeeeeeee2](https://user-images.githubusercontent.com/79726511/143480896-6fdc23ce-0b36-4268-b676-e3f1fd26f3e3.png)

3) but i did not get it when i saved my setting, and i only get the Created filter, as follow:-
![rrrrrrrrrrrrrrrrrrrrrrrrrr3](https://user-images.githubusercontent.com/79726511/143480904-2fba4f8e-5b96-4965-9d74-999b36010770.png)

do you have any idea why this is not working for me?
Thanks

**Reply by wobba** [reply]
![image](https://user-images.githubusercontent.com/51104/143491527-35585c93-921e-451d-a4f4-54ec5276a825.png)
